src/coreclr/nativeaot/System.Private.CoreLib/src/System/Array.NativeAot.cs

```diff
@@ -1196,6 +1199,68 @@ public void RemoveAt(int index)
         }
     }
 
+    internal class SZGenericArrayEnumeratorBase : IDisposable
+    {
+        protected int _index;
+        protected int _endIndex;
+
+        internal SZGenericArrayEnumeratorBase()
+        {
+            _index = -1;
+        }
+
+        public bool MoveNext()
+        {
+            if (_index < _endIndex)
+            {
+                _index++;
+                return (_index < _endIndex);
+            }
+            return false;
+        }
+
+        public void Dispose()
+        {
+        }
+    }
+
+    internal sealed class SZGenericArrayEnumerator<T> : SZGenericArrayEnumeratorBase, IEnumerator<T>
+    {
+        private readonly T[] _array;
+
+        // Passing -1 for endIndex so that MoveNext always returns false without mutating _index
+        internal static readonly SZGenericArrayEnumerator<T> Empty = new SZGenericArrayEnumerator<T>(null, -1);
+
+        internal SZGenericArrayEnumerator(T[] array, int endIndex)
+        {
+            _array = array;
+            _endIndex = endIndex;
+        }
+
+        public T Current
+        {
+            get
+            {
+                if ((uint)_index >= (uint)_endIndex)
+                    ThrowHelper.ThrowInvalidOperationException();
+                return _array[_index];
+            }
+        }
+
+        object IEnumerator.Current
```

Comment:
Is this style change now preferred to the [other way in line 143](https://github.com/dotnet/runtime/pull/88371/files#diff-fe6fc3f88a51a2089f4703b7cf1cb695a440fef64a2cf78feed4417f11220a07R143)?
```csharp
     object? IEnumerator.Current => Current;
```